shared-module/audiomixer/Mixer.c

```diff
@@ -103,11 +103,28 @@ void audiomixer_mixer_reset_buffer(audiomixer_mixer_obj_t *self,
 
 __attribute__((always_inline))
 static inline uint32_t add16signed(uint32_t a, uint32_t b) {
+    #if (defined(__ARM_ARCH_7EM__) && (__ARM_ARCH_7EM__ == 1))  // Cortex-M4 w/FPU
     return __QADD16(a, b);
+    #else
+    uint32_t result = 0;
+    for (int8_t i = 0; i < 2; i++) {
+        int16_t ai = a >> (sizeof(int16_t) * 8 * i);
+        int16_t bi = b >> (sizeof(int16_t) * 8 * i);
+        int32_t intermediate = (int32_t)ai + bi / 2;
```

Comment:
Can you say why bi is divided by 2 here?

Reply:
Not sure why that was in the code I copied but it was causing an error so removed now. Thanks.

Reply:
Is there a casting error possible? (I am not sure it matters.)

I was not completely sure which range gets saturated to be honest either.

https://www.keil.com/support/man/docs/armasm/armasm_dom1361289886623.htm